core/tx_pool.go

```diff
@@ -526,10 +527,12 @@ func (pool *TxPool) stats() (int, int) {
 // StatsWithMinBaseFee retrieves the current pool stats, namely the number of pending and the
 // number of queued (non-executable) transactions greater equal minBaseFee.
 func (pool *TxPool) StatsWithMinBaseFee(minBaseFee *big.Int) (int, int) {
+	statsStart := time.Now()
 	pool.mu.RLock()
```

Comment:
this is unrelated to this PR, but I guess we should've acquired the write lock here. `statsWithMinBaseFee` requires to sort the txns, therefore it actually "writes" to the pool. @jonastheis am I missing something here?

Reply:
original `Stats`only acquires the read lock because it doesn't do any modifications.

Reply:
> statsWithMinBaseFee requires to sort the txns, therefore it actually "writes" to the pool
you mean the `flatten` function?

Reply:
original code 
```
func (pool *TxPool) stats() (int, int) {
	pending := 0
	for _, list := range pool.pending {
		pending += list.Len()
	}
	queued := 0
	for _, list := range pool.queue {
		queued += list.Len()
	}
	return pending, queued
}
```

Reply:
yeah, maybe @omerfirmak is right, search the usages,  all use the write lock

Reply:
Yep, I think we should acquire the write lock. My bad!

Reply:
changed [ef87c11](https://github.com/scroll-tech/go-ethereum/pull/983/commits/ef87c11f731747fe9cfedec87ab1ebbec6f58349)